lib/SILGen/SILGenApply.cpp

```diff
@@ -3695,11 +3697,20 @@ CallEmission::applyNormalCall(SGFContext C) {
 
   auto mv = callee.getFnValue(SGF, isCurried, borrowedSelf);
 
+  // Materialize for set could temporarily escape its arguments.
+  if (callee.isMaterializeForSet) {
```

Comment:
It seems unfortunate that we’d have to call out materializeForSet by name here. Is it because of the callback nature of materializeForSet’s writeback, which is breaking some rule that user code would not be permitted to break?

Reply:
Yes, materializeForSet can temporarily escape ``@noescape`` arguments until the writeback.

```
protocol P {
  associatedtype Element
  subscript<U>(a: (Element) -> U, b: (U) -> Element) -> U { get set }
}

func increment<T : P>(p: inout T) where T.Element == String {
  p[{Int($0)!}, {String($0)}] += 1
}
```

I can rename the variable to something else. However, ``materializeForSet`` is the only instance of ``canEscapeArgumentsUntilWriteback`` that I am aware of.